zerver/tests/test_sessions.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+
+from zerver.lib.sessions import (
+    user_sessions,
+    delete_session,
+    delete_user_sessions,
+    delete_realm_user_sessions,
+    delete_all_user_sessions,
+    delete_all_deactivated_user_sessions,
+)
+
+from zerver.models import (
+    UserProfile, get_user_profile_by_id,
+    get_user_profile_by_email, get_realm
+)
+
+from zerver.lib.test_classes import ZulipTestCase
+
+
+class TestSessions(ZulipTestCase):
+    def test_delete_session(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        for session in user_sessions(user_profile):
+            delete_session(session)
+        result = self.client_get("/")
+        self.assertEqual('/login', result.url)
+
+    def test_delete_user_session(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        delete_user_sessions(user_profile)
+        result = self.client_get("/")
+        self.assertEqual('/login', result.url)
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        self.login('othello@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        delete_user_sessions(user_profile)
+        self.assertIn('_auth_user_id', self.client.session)
+
+    def test_delete_realm_user_sessions(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        realm = get_realm('zulip')
+        self.login('othello@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        delete_realm_user_sessions(realm)
+        result = self.client_get('/')
+        self.assertEqual('/login', result.url)
+
+    def test_delete_all_user_sessions(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        self.login('othello@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        delete_all_user_sessions()
+        result = self.client_get('/')
+        self.assertEqual('/login', result.url)
```

Comment:
I think the structure of these tests isn't quite right, because when you do `self.login("othello@zulip.com")` here, it replaces the hamlet login.  What you want to do instead is something like this:

```
self.login('hamlet@zulip.com')
self.assertIn('_auth_user_id', self.client.session)
￼delete_all_user_sessions()
result = self.client_get('/')
self.assertEqual('/login', result.url)

self.login('othello@zulip.com')
self.assertIn('_auth_user_id', self.client.session)
￼delete_all_user_sessions()
result = self.client_get('/')
self.assertEqual('/login', result.url)
```

Basically where you test with both users.  And then for e.g. `test_delete_realm_users_sessions`, you'd do the same thing, except the 2 users would be in different realms (or it'd be the same user each time but 2 different realms used), so you'd confirm only the user in the realm got removed.

To make this clean, you probably want to write a test helper function that you could use like this:

```
def test_delete_all_user_sessions(self):
    self.do_test_session(hamlet@zulip.com, lambda: delete_all_user_sessions(), True)
    self.do_test_session(sipbtest@mit.edu, lambda: delete_all_user_sessions(), True)

def test_delete_realm_user_sessions(self):
    self.do_test_session(hamlet@zulip.com, lambda: delete_all_user_sessions(), True)
    self.do_test_session(sipbtest@mit.edu, lambda: delete_all_user_sessions(), False)
```

to avoid a lot of code duplication in these tests.